add_keys.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+
+import os
+import json
+import logging
+import sys
+import time
+import click
+from web3 import Web3, WebsocketProvider, HTTPProvider
+from web3.exceptions import SolidityError, CannotHandleRequest, TimeExhausted
+
+logger = logging.getLogger().setLevel("INFO")
+
+ARTIFACTS_DIR = './assets'
+POOL_ADDRESS = "0xae7ab96520de3a18e5e111b5eaab095312d7fe84"
+POOL_ARTIFACT_FILE = 'Lido.json'
+REGISTRY_ARTIFACT_FILE = 'NodeOperatorsRegistry.json'
+GAS_LIMIT=13500000
+
+
+def init_provider(eth1_provider):
+    '''
+    Initialise the Web3 provider.
+    '''
+    if eth1_provider.startswith('http'):
+        provider = HTTPProvider(eth1_provider)
+    elif eth1_provider.startswith('ws'):
+        provider = WebsocketProvider(eth1_provider)
+    else:
+        logging.error('Unsupported ETH provider!')
+        sys.exit(1)
+
+    return Web3(provider)
+
+
+def init_contracts(w3: Web3, pool_address):
+    '''
+    Initialise contracts.
+    '''
+
+    if not Web3.isChecksumAddress(pool_address):
+        pool_address = Web3.toChecksumAddress(pool_address)
+
+    pool_abi_path = os.path.join(ARTIFACTS_DIR, POOL_ARTIFACT_FILE)
+    registry_abi_path = os.path.join(ARTIFACTS_DIR, REGISTRY_ARTIFACT_FILE)
+
+    # Get Pool contract
+    with open(pool_abi_path, 'r') as file:
```

Comment:
The encoding that this uses [depends on the environment](https://docs.python.org/3/library/functions.html?highlight=open#open). In practice it is usually UTF-8, but to be sure, we can be explicit about it and write `open(fname, 'r', encoding='utf-8')` to sidestep potential issues.